src/lib.rs

```diff
@@ -232,6 +252,30 @@ impl<T> Tree<T> {
         self.vec.extend(other_tree.vec);
         unsafe { self.get_unchecked_mut(other_tree_root_id) }
     }
+
+    /// Maps a `Tree<T>` to `Tree<U>` by applying a function to all node values,
+    /// copying over the tree's structure and node ids untouched, consuming `self`.
+    pub fn map<U>(self, mut transform: impl FnMut(T) -> U) -> Tree<U> {
```

Comment:
```suggestion
    pub fn map<F, U>(self, mut transform: F) -> Tree<U> where F: FnMut(T) -> U {
```

So `F` can be specified via turbofish. (Same for the other three signatures.)

Reply:
Done.